subprojects/docs/src/snippets/kotlinDsl/sourceSetConvention/kotlin/build.gradle.kts

```diff
@@ -6,7 +6,7 @@ plugins {
 sourceSets {
     main {
         withConvention(GroovySourceSet::class) {
-            groovy.srcDir("src/core/groovy")
+            groovy.setSrcDirs(listOf("src/core/groovy"))
```

Comment:
This is doing something a little different.  Does `groovy.srcDir` not work?

`groovy.srcDir("src/core/groovy")` adds the directory src/core/groovy to the set of source directories (on top of the built-in ones).  `setSrcDirs` replaces the set of source directories entirely.

This snippet is referenced in the Kotlin DSL chapter and it mentions `withConvention`/Conventions in general: 
https://github.com/gradle/gradle/blob/master/subprojects/docs/src/docs/userguide/api/kotlin_dsl.adoc#about-conventions

So we may be able to just get rid of this entirely.

Reply:
Well, my idea is to update the documentation when we deprecate/remove `withConvention` entirely. There still might be users out there who need this information at the moment.